### ISSUE TITLE
Add field-definition-expression of virtual fields as default expression

### ISSUE THREAD

**Issue by koenigherford** [qgis/QGIS#43644]
**Feature description.**
Virtual fields are commonly used to calculate and display values on the fly. 

If a virtual field is used in a form the user may expect to have a calculated value in this field already when he first opens the form by digitizing a new feature. To achieve this he has to copy the field-definition-expression into the default-value-expression.
This will produce redundant code. 

It will be nice to use in virtual fields the actual field-definition-expression as default-value-expression to reduce maintainance efforts for the user and sources of errors.

**Comment by signedav** [comment]
I don't think it's intended to have the possibility to override virtual fields with default value expressions "applied on update". So the default value expressions should not be considered in virtual fields at all.

Instead it would be great to have the virtual fields calculated on the fly with their own expressions.

@koenigherford Let's discuss it out here #41243 and then update this issue or close it. Okay?

**Comment by koenigherford** [comment]
Yes, it would be great to have the virtual fields calculated on the fly with their own expressions. 

This will allow to work without default values in virtual fields. In this case using virtual fields the user shall maybe prevented  to create default values by disabling "default-values"-settings-forms from QGIS-side
